### PR TITLE
pulseaudio: bump dependencies + modernize more for conan v2

### PR DESCRIPTION
**Pull request by SpaceIm**
closes https://github.com/conan-io/conan-center-index/issues/16493

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by jwillikers — APPROVED** [review]

**Review by StellaSmith — APPROVED** [review]

**Review by Croydon — DISMISSED** [review]

**Comment by SpaceIm** [comment]
```
ERROR: Missing binary: libsndfile/1.2.0:35b74efdc6da71570a7a5ab8f29b5da604122183
libsndfile/1.2.0: WARN: Can't find a 'libsndfile/1.2.0' package for the specified settings, options and dependencies:
- Settings: arch=x86_64, build_type=Release, compiler=gcc, compiler.libcxx=libstdc++11, compiler.version=11, os=Linux
- Options: experimental=False, fPIC=True, programs=True, shared=False, with_alsa=False, with_external_libs=True, with_mpeg=True, flac:fPIC=True, flac:shared=False, libmp3lame:fPIC=True, libmp3lame:shared=False, mpg123:fPIC=True, mpg123:flexible_resampling=True, mpg123:icy=True, mpg123:id3v2=True, mpg123:ieeefloat=True, mpg123:layer1=True, mpg123:layer2=True, mpg123:layer3=True, mpg123:module=dummy, mpg123:moreinfo=True, mpg123:network=True, mpg123:seektable=1000, mpg123:shared=False, ogg:fPIC=True, ogg:shared=False, opus:fPIC=True, opus:fixed_point=False, opus:shared=False, vorbis:fPIC=True, vorbis:shared=False
- Dependencies: ogg/1.3.5, vorbis/1.3.7, flac/1.4.2, opus/1.3.1, mpg123/1.29.3, libmp3lame/3.100
- Requirements: flac/1.Y.Z, libmp3lame/3.Y.Z, mpg123/1.Y.Z, ogg/1.Y.Z, opus/1.Y.Z, vorbis/1.Y.Z
- Package ID: 35b74efdc6da71570a7a5ab8f29b5da604122183

ERROR: Missing prebuilt package for 'libsndfile/1.2.0'
```

@danimtb @uilianries libsndfile v1 packages are still missing.

**Review by jwillikers — DISMISSED** [review]

**Comment by SpaceIm** [comment]
I've opened https://github.com/conan-io/conan-center-index/pull/16517 to generate missing libsndfile packages.

**Review by jwillikers — APPROVED** [review]

**Comment by SpaceIm** [comment]
```
ERROR: Missing binary: gettext/0.21:e826dcf4aca2bbe8d1ec5b712ce0ab0b6a97a2ed

gettext/0.21: WARN: Can't find a 'gettext/0.21' package binary 'e826dcf4aca2bbe8d1ec5b712ce0ab0b6a97a2ed' for the configuration:
[settings]
arch=x86_64
build_type=Release
os=Linux
[requires]
libiconv/1.17.Z

ERROR: Missing prebuilt package for 'gettext/0.21'
Check the available packages using 'conan list gettext/0.21:* -r=remote'
or try to build locally from sources using the '--build=gettext/0.21' argument
```

@danimtb @uilianries @jcar87 gettext v2 packages are missing, but there was an effort in https://github.com/conan-io/conan-center-index/pull/16070 to migrate this one to conan v2 during CI maintenance. It seems that packages have not been generated 3 weeks ago because m4 packages were missing. Maybe it's better now?

File: recipes/pulseaudio/all/conanfile.py [hunk 1]
```diff
@@ -52,26 +53,29 @@ def configure(self):
...
+    def layout(self):
+        basic_layout(self, src_folder="src")
+
     def requirements(self):
         self.requires("libiconv/1.17")
-        self.requires("libsndfile/1.0.31")
-        self.requires("libcap/2.65")
+        self.requires("libsndfile/1.2.0")
+        self.requires("libcap/2.66")
         self.requires("libtool/2.4.7")
```
Comment (prince-chrismc) [inline recipes/pulseaudio/all/conanfile.py:63]: Why is libtool a requires?

Comment (SpaceIm) [reply recipes/pulseaudio/all/conanfile.py:63]: for libltdl

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**